instrumentation/rocketmq/src/main/java/brave/rocketmq/clients/SendMessageBraveHookImpl.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2013-2024 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.rocketmq.clients;
+
+import brave.Span;
+import org.apache.rocketmq.client.hook.SendMessageContext;
+import org.apache.rocketmq.client.hook.SendMessageHook;
+import org.apache.rocketmq.client.impl.CommunicationMode;
+import org.apache.rocketmq.client.producer.SendResult;
+import org.apache.rocketmq.client.producer.SendStatus;
+import org.apache.rocketmq.common.message.Message;
+
+
+public class SendMessageBraveHookImpl implements SendMessageHook {
+
+  final RocketMQTracing tracing;
+
+  public SendMessageBraveHookImpl(RocketMQTracing tracing) {
+    this.tracing = tracing;
+  }
+
+  @Override
+  public String hookName() {
+    return "SendMessageBraveHook";
+  }
+
+  @Override
+  public void sendMessageBefore(SendMessageContext context) {
+    if (context == null || context.getMessage() == null) {
+      return;
+    }
+
+    Message msg = context.getMessage();
+    MessageProducerRequest request = new MessageProducerRequest(msg);
+    Span span =
+      SpanUtil.createAndStartSpan(tracing, tracing.producerExtractor, tracing.producerSampler,
+        request,
+        msg.getProperties());
+    span.name(TraceConstants.TO_PREFIX + msg.getTopic());
```

Comment:
look at the other messaging implementations, and notice that we do not add any non-default tags to the span. That's what customization is for, and it also significantly reduces the amount of code and constants to review. 

Similar to KafkaTags start only with very very minimal, like pick 2 tags and delete everything else from this PR that isn't required to pass similarly